packages/core/content-releases/admin/src/components/ReleaseModal.tsx

```diff
@@ -51,24 +95,130 @@ export const ReleaseModal = ({
       </ModalHeader>
       <Formik
         validateOnChange={false}
-        onSubmit={handleSubmit}
-        initialValues={initialValues}
+        onSubmit={(values) => {
+          handleSubmit({
+            ...values,
+            scheduledAt: values.isScheduled ? getScheduledTimestamp(values) : null,
+          });
+        }}
+        initialValues={{
+          ...initialValues,
+          timezone: initialValues.timezone ? initialValues.timezone : currentTimezone.value,
+        }}
         validationSchema={RELEASE_SCHEMA}
       >
-        {({ values, errors, handleChange }) => (
+        {({ values, errors, dirty, handleChange, setFieldValue }) => (
           <Form>
             <ModalBody>
-              <TextInput
-                label={formatMessage({
-                  id: 'content-releases.modal.form.input.label.release-name',
-                  defaultMessage: 'Name',
-                })}
-                name="name"
-                value={values.name}
-                error={errors.name}
-                onChange={handleChange}
-                required
-              />
+              <Flex direction="column" alignItems="stretch" gap={6}>
+                <TextInput
+                  label={formatMessage({
+                    id: 'content-releases.modal.form.input.label.release-name',
+                    defaultMessage: 'Name',
+                  })}
+                  name="name"
+                  value={values.name}
+                  error={errors.name}
+                  onChange={handleChange}
+                  required
+                />
+                <Checkbox
+                  name="isScheduled"
+                  value={values.isScheduled}
+                  onChange={(event) => {
+                    setFieldValue('isScheduled', event.target.checked);
+                    if (!event.target.checked) {
+                      // Clear scheduling info from a release on unchecking schedule release
+                      setFieldValue('date', null);
+                      setFieldValue('time', '');
+                    } else {
+                      setFieldValue('date', initialValues.date);
+                      setFieldValue('time', initialValues.time);
+                    }
```

Comment:
I find this a bit weird, why are doing this here?

Reply:
On ticking/unticking I need to keep track of dirty state and so refill to initial data if ticked back to its original state.
https://github.com/strapi/strapi/assets/7886244/afa7c5ed-b288-4d7f-9b9d-aec523be486f

Reply:
https://github.com/strapi/strapi/assets/7886244/72d9f7b6-26c3-4b98-a7b0-2c0d7f7c8883